### ISSUE TITLE
No files changed Dependabot Error with GitHub Actions

### ISSUE THREAD

**Issue by stephenmontgomery** [dependabot/dependabot-core#6634]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package ecosystem

Github actions

### dependabot.yml content

```yaml
# To get started with Dependabot version updates, you'll need to specify which
# package ecosystems to update and where the package manifests are located.
# Please see the documentation for all configuration options:
# https://docs.github.com/github/administering-a-repository/configuration-options-for-dependency-updates

version: 2
updates:
  # Automated updates for Dockerfile dependencies
  - package-ecosystem: maven
    directory: "/"
    allow:
      - dependency-name: "com.contrastsecurity:contrast-agent"
      - dependency-name: "com.datadoghq:dd-java-agent"
    # Target branch is needed as a workaround for Dependabot's requirement
    # that each combination of package-ecosystem/directory/target-branch be unique
    target-branch: main
    schedule:
      interval: daily
      time: "10:00"
    pull-request-branch-name:
      separator: "-"
    labels:
      - "dependencies"
    open-pull-requests-limit: 2
  - package-ecosystem: maven
    registries: "*"
    directory: "/"
    ignore:
      # contrast-agent and dd-java-agent are downloaded in the configuration above
      # contrast-agent must be search for updates only in public repositories
      - dependency-name: "com.contrastsecurity:contrast-agent"
      - dependency-name: "com.datadoghq:dd-java-agent"
      - dependency-name: "*"
        update-types: [ "version-update:semver-major" ]
    schedule:
      interval: daily
      time: "11:00"
    pull-request-branch-name:
      separator: "-"
    labels:
      - "dependencies"
    open-pull-requests-limit: 6
  - package-ecosystem: "github-actions"
    directory: "/"
    labels:
      - "gh-dependencies"
    schedule:
      interval: "daily"
      time: "12:00"
    pull-request-branch-name:
      separator: "-"
    open-pull-requests-limit: 3
```

### Updated dependency

convictional/trigger-workflow-and-wait 1.6.1 -> 1.6.5

### What you expected to see, versus what you actually saw

- A new DP PR raised to update the GH action file with the new version of convictional/trigger-workflow-and-wait

### Smallest manifest that reproduces the issue

- Have DP enabled/scheduled in our GitHub Repo
- Currently failing

![image](https://user-images.githubusercontent.com/5411859/218069964-1015bb4e-aa30-4f8c-941b-1b2517728614.png)

- Error is: 
``` updater | INFO <job_600866772> Checking if convictional/trigger-workflow-and-wait 1.6.1 needs updating
  proxy | 2023/02/09 15:47:45 [056] GET https://github.com:443/convictional/trigger-workflow-and-wait.git/info/refs?service=git-upload-pack
  proxy | 2023/02/09 15:47:45 [056] * authenticating git server request (host: github.com)
  proxy | 2023/02/09 15:47:45 [056] 200 https://github.com:443/convictional/trigger-workflow-and-wait.git/info/refs?service=git-upload-pack
updater | INFO <job_600866772> Latest version is 1.6.5
updater | INFO <job_600866772> Requirements to unlock own
updater | INFO <job_600866772> Requirements update strategy 
updater | INFO <job_600866772> Updating convictional/trigger-workflow-and-wait from 1.6.1 to 1.6.5
updater | I, [2023-02-09T15:47:45.974200 #7]  INFO -- sentry: ** [Raven] Sending event 742e87bbac8f494baea9a5f72594757a to Sentry
  proxy | 2023/02/09 15:47:46 [058] POST https://sentry.io:443/api/1451818/store/
  proxy | 2023/02/09 15:47:46 [058] 200 https://sentry.io:443/api/1451818/store/
updater | ERROR <job_600866772> Error processing convictional/trigger-workflow-and-wait (RuntimeError)
updater | ERROR <job_600866772> No files changed!
updater | ERROR <job_600866772> /home/dependabot/github_actions/lib/dependabot/github_actions/file_updater.rb:28:in `updated_dependency_files'
updater | ERROR <job_600866772> /home/dependabot/dependabot-updater/lib/dependabot/updater.rb:747:in `generate_dependency_files_for'
updater | ERROR <job_600866772> /home/dependabot/dependabot-updater/lib/dependabot/updater.rb:311:in `check_and_create_pull_request'
updater | ERROR <job_600866772> /home/dependabot/dependabot-updater/lib/dependabot/updater.rb:109:in `check_and_create_pr_with_error_handling'
updater | ERROR <job_600866772> /home/dependabot/dependabot-updater/lib/dependabot/updater.rb:82:in `block in run'
updater | ERROR <job_600866772> /home/dependabot/dependabot-updater/lib/dependabot/updater.rb:82:in `each'
updater | ERROR <job_600866772> /home/dependabot/dependabot-updater/lib/dependabot/updater.rb:82:in `run'
updater | ERROR <job_600866772> /home/dependabot/dependabot-updater/lib/dependabot/update_files_job.rb:17:in `perform_job'
updater | ERROR <job_600866772> /home/dependabot/dependabot-updater/lib/dependabot/base_job.rb:50:in `run'
updater | ERROR <job_600866772> bin/update_files.rb:23:in `<main>'
```

- Googled and found https://github.com/dependabot/dependabot-core/issues/1609 - looks similar behaviour (but for Docker) as we had 2 separate GH Action files in the same repo - one of them was patched to the latest version of trigger-workflow-and-wait (1.6.5) and the other file wasn't (package-service.yml). The reason for this discrepancy was that out of date file originally lived in a diff GH repo (and referenced) and was then copied manually into the repo recently in a PR.

<img width="1211" alt="image" src="https://user-images.githubusercontent.com/5411859/218071878-dca95109-06e8-4b2a-a219-68e661e0e2c8.png">

- Manually raising a PR to bring package-service.yml's trigger-workflow-and-wait up to 1.6.5 (or version hash) and merging fixes DP: 

```
updater | INFO <job_600866772> Checking if convictional/trigger-workflow-and-wait 1.6.5 needs updating
  proxy | 2023/02/10 10:20:00 [044] GET https://github.com:443/convictional/trigger-workflow-and-wait.git/info/refs?service=git-upload-pack
  proxy | 2023/02/10 10:20:00 [044] * authenticating git server request (host: github.com)
  proxy | 2023/02/10 10:20:00 [044] 200 https://github.com:443/convictional/trigger-workflow-and-wait.git/info/refs?service=git-upload-pack
updater | INFO <job_600866772> Latest version is 1.6.5
updater | INFO <job_600866772> No update needed for convictional/trigger-workflow-and-wait 1.6.5
```

**Comment by deivid-rodriguez** [comment]
I see!

I think this is a duplicate of https://github.com/dependabot/dependabot-core/issues/3750, so please track that other issue. It even has an open PR linked to it, so I hope we can fix it soon-ish!

**Comment by d-e-s-o** [comment]
So both this issue and #3750 have been closed, but the issue still persists, no? I've encountered the same `RuntimeError` and it got resolved only after unifying the versions of dependencies.
